friprosveta/views.py

```diff
@@ -478,6 +479,44 @@ def allocations_json(request, timetable_slug=None):
     filtered_allocations = _allocation_set(param_ids,
                                            tt.allocations,
                                            request.user.is_staff)
+
+    #returns more information about allocations
+    request_r = request.GET
+    if 'mode' in request_r:
+        mode_l = request_r['mode']
+        if mode_l == 'ext':
+            multiple_allocations = namedtuple('AllocationVM', ['object', 'day_index', 'hour_index', 'duration'])
```

Comment:
After some consideration I believe using namedtuple is not necessary.

Instead add properties hour_index and day_index directly to the Allocation model

https://github.com/UL-FRI/urnik/blob/4f2f303f429bad2db0685ec9ab18d51b090ce55a/timetable/models.py#L598 

and use allocation object directly.

Reply:
Added properties directly to the object.